File: android/src/main/java/com/lightningkite/khrysalistemplate/layouts/DateRangeDemoXml.kt
//
// DateRangeDemoXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.khrysalistemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysalistemplate.R

class DateRangeDemoXml {

    lateinit var month: com.lightningkite.khrysalis.views.CustomView
    lateinit var month2: com.lightningkite.khrysalis.views.CustomView
    lateinit var month3: com.lightningkite.khrysalis.views.CustomView
    lateinit var month4: com.lightningkite.khrysalis.views.CustomView
    lateinit var monthDelegate: com.lightningkite.khrysalis.views.MonthCVD
    lateinit var month2Delegate: com.lightningkite.khrysalis.views.SelectDateMonthCVD
    lateinit var month3Delegate: com.lightningkite.khrysalis.views.SelectDateRangeMonthCVD
    lateinit var month4Delegate: com.lightningkite.khrysalis.views.SelectMultipleDatesMonthCVD
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.date_range_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        month = view.findViewById<com.lightningkite.khrysalis.views.CustomView>(R.id.month)
        month2 = view.findViewById<com.lightningkite.khrysalis.views.CustomView>(R.id.month2)
        month3 = view.findViewById<com.lightningkite.khrysalis.views.CustomView>(R.id.month3)
        month4 = view.findViewById<com.lightningkite.khrysalis.views.CustomView>(R.id.month4)
        monthDelegate = view.findViewById<CustomView>(R.id.month).delegate as com.lightningkite.khrysalis.views.MonthCVD
        month2Delegate = view.findViewById<CustomView>(R.id.month2).delegate as com.lightningkite.khrysalis.views.SelectDateMonthCVD
        month3Delegate = view.findViewById<CustomView>(R.id.month3).delegate as com.lightningkite.khrysalis.views.SelectDateRangeMonthCVD
        month4Delegate = view.findViewById<CustomView>(R.id.month4).delegate as com.lightningkite.khrysalis.views.SelectMultipleDatesMonthCVD
        
        
        return view
    }
}

File: ios/KhrysalisTemplate/swiftResources/layouts/DateRangeDemoXml.swift
//
// DateRangeDemoXml.swift
// Created by Khrysalis XML
//

import UIKit
import Khrysalis
import MapKit

public class DateRangeDemoXml {
    
    public unowned var xmlRoot: UIView!
    public func setup(dependency: ViewDependency) -> UIView {
        let view = ScrollViewVertical(frame: .zero)
        view.addSubview(LinearLayout(frame: .zero)) { view in 
            view.orientation = .y
            view.padding = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
            view.gravity = .topLeft
            
            view.addSubview(
                CustomView(frame: .zero),
                minimumSize: CGSize(width: 0, height: 0),
                size: CGSize(width: 0, height: 350.0),
                margin: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 40.0),
                padding: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0),
                gravity: .topFill,
                weight: 0
            ) { view in 
                self.month = view
                applyColor(view, R.color.white) { (c, s) in
                    view.backgroundColor = c
                }
                let dg = MonthCVD()
                view.delegate = dg
                self.monthDelegate = dg
            }
            
            view.addSubview(
                CustomView(frame: .zero),
                minimumSize: CGSize(width: 0, height: 0),
                size: CGSize(width: 0, height: 350.0),
                margin: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 40.0),
                padding: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0),
                gravity: .topFill,
                weight: 0
            ) { view in 
                self.month2 = view
                applyColor(view, R.color.white) { (c, s) in
                    view.backgroundColor = c
                }
                let dg = SelectDateMonthCVD()
                view.delegate = dg
                self.month2Delegate = dg
            }
            
            view.addSubview(
                CustomView(frame: .zero),
                minimumSize: CGSize(width: 0, height: 0),
                size: CGSize(width: 0, height: 350.0),
                margin: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 40.0),
                padding: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0),
                gravity: .topFill,
                weight: 0
            ) { view in 
                self.month3 = view
                applyColor(view, R.color.white) { (c, s) in
                    view.backgroundColor = c
                }
                let dg = SelectDateRangeMonthCVD()
                view.delegate = dg
                self.month3Delegate = dg
            }
            
            view.addSubview(
                CustomView(frame: .zero),
                minimumSize: CGSize(width: 0, height: 0),
                size: CGSize(width: 0, height: 350.0),
                margin: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 40.0),
                padding: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0),
                gravity: .topFill,
                weight: 0
            ) { view in 
                self.month4 = view
                applyColor(view, R.color.white) { (c, s) in
                    view.backgroundColor = c
                }
                let dg = SelectMultipleDatesMonthCVD()
                view.delegate = dg
                self.month4Delegate = dg
            }
            
        }
        xmlRoot = view
        return view
    }
    
    public unowned var month2Delegate: SelectDateMonthCVD!
    public unowned var month3Delegate: SelectDateRangeMonthCVD!
    public unowned var monthDelegate: MonthCVD!
    public unowned var month4Delegate: SelectMultipleDatesMonthCVD!
    public unowned var month2: CustomView!
    public unowned var month3: CustomView!
    public unowned var month: CustomView!
    public unowned var month4: CustomView!
    
}
